spinalcordtoolbox/scripts/sct_image.py

```diff
@@ -324,6 +340,27 @@ def main(argv: Sequence[str]):
         dim = dim_list.index(dim)
         im_out = split_data(im_in, dim)
 
+    elif arguments.stitch:
+        # preserve original orientation (we assume it's consistent among all images)
+        orig_ornt = im_in_list[0].orientation
+
+        # reorient input files and save them to a temp directory
+        path_tmp = tmp_create(basename="image-stitching")
+        fnames_in = []
+        for file in arguments.i:
+            temp_file_path = os.path.join(path_tmp, os.path.basename(file))
+            im_in = Image(file)
+            im_in_rpi = change_orientation(im_in, 'RPI')
+            im_in_rpi.save(temp_file_path, dtype=output_type, verbose=verbose)
+            fnames_in.append(temp_file_path)
+
+        # order fs_names in descending order based on dimensions (largest -> smallest)
+        fnames_in_sorted = sorted(fnames_in, key=lambda fname: max(Image(fname).dim), reverse=True)
+
+        # stitch and reorient
+        im_out_rpi = stitch_images(fnames_in=fnames_in_sorted, fname_out=os.path.join(path_tmp, 'stitched.nii.gz'))
+        im_out = [change_orientation(im_out_rpi, orig_ornt)]
```

Comment:
On second thought, I think that maybe I like the idea of doing the orientation and sorting inside the `stitch_images` function itself. (That way, in theory, the `stitch_images` API function can be called on any images, without any prerequisites.)

Reply:
It might also be worth changing `stitch_images` so that it accepts `Image` objects, rather than filenames?

Reply:
> The only thing I'm not sure of is how the RPI reorientation affects the QC report. If we're not passing RPI-oriented images to the QC report, will it still display correctly?

@joshuacwnewton I think the RPI orientation should not affect / should not be necessary for the quality check - because we are converting the stitched images back to the original orientation anyway? Thus, having the non-stitched images in the original orientation should complement the final, stitched image. The conversion to RPI is only necessary because `stitching` (c++) expects the RPI orientation :slightly_smiling_face:

Reply:
Yes, the fact that stitching requires a particular orientation seems like an implementation detail of `isct_stitching`, that we can maybe shield from the users of the `stitch_images()` function.